### ISSUE TITLE
Module overwriting my custom Name tag value

### ISSUE THREAD

**Issue by EdyKnopfler** [terraform-aws-modules/terraform-aws-rds#353]
## Is your request related to a new offering from AWS?
No.

## Is your request related to a problem? Please describe.
I'm trying to do something like this:

```terraform
module "db" {
  source = "terraform-aws-modules/rds/aws"
  version = "~> 3.3.0"

  ...

  db_instance_tags = {
    Name = "my_custom_Name_tag"  # this!
  }
}
```

But inside `modules` directory, the code merges my custom tags with a self-defined `Name` tag, like below, and this overwrites my custom value:

```terraform
  # taken from modules/db_instance/main.tf, occurs in other modules as well
  tags = merge(
    var.tags,
    {
      "Name" = format("%s", var.identifier)
    },
  )
```

## Describe the solution you'd like.
Just changing the order of tags in merge function would solve the problem, wouldn't?

```terraform
  tags = merge(
    {
      "Name" = format("%s", var.identifier)   # default tag comes first
    },
    var.tags,
  )
```
## Describe alternatives you've considered.
The only possible is not to use the module, mas I really want to use it.

## Additional context
No, thanks. I hope this suffices.

**Comment by bryantbiggs** [comment]
there are a few things to unpack here:
1. the name is set by `var.identifier` and that is passed to the RDS API and is the name you see show up in the console; simply changing the name tag will NOT change the name you see in the console 
2. Tag precedence is evaluated as an overloaded operation - tags that come first can be overloaded/overwritten by tags of the same key later in the merge order. This is why "default" tags (commonly `var.tags`) come first in the merge operation, with any specific/custom tags coming later so that users get a "sane" set of "default" tags but with flexibility to overload/overwrite if desired

tl;dr - the `Name` tag here is not responsible for the function you intended; `var.identifier` is
